imports/plugins/core/graphql/server/schemas/cart.graphql

```diff
@@ -0,0 +1,364 @@
+  # Enforce database uniqueness such that an account only has one cart.
+  "The account that owns the cart."
+  account: Account
+
+  # Unused until checkout implementation.
+  #billing: [Payment]! # TODO: What is this?
+
+  "TimeStamp at which the entity was created."
+  createdAt: DateTime!
+
+  # We want to change discounts but keeping this for now
+  # Unused until checkout implementation.
+  # "The total discount amount for the cart"
+  #discount: Float
+
+  # Only necessary until making an order. Order should take from here or from account email for user cart
+  # Unused until checkout implementation.
+  # "The email associated with the cart. For anonymous checkout"
+  #email: Email
+
+  "The items that have been added to the cart."
+  items: [CartItem]!
+
+  # session: Session! # Remove - Moving to sessionless API.
+
+  # Unused until checkout implementation.
+  #shipping: [Shipment]! # TODO: What is this?
+
+  "The shop associated with the cart."
+  shop: Shop!
+
+  # TODO: Is this the total tax? Calculated at runtime, or flattened?
+  # Unused until checkout implementation.
+  # "The total tax associated with the cart."
+  #tax: Float
+
+  # Unused until checkout implementation.
+  #taxRatesByShop: JSONObject # How is this used?
+
+  # Unused until checkout implementation.
+  #taxes: [JSONObject] # TODO: What is this?
+
+  # Undecided if token is necessary.
+  #"Secret token for anonymous carts, a token needed for requests"
+  #token: String
+
+  "TimeStamp at which the entity was last updated."
+  updatedAt: DateTime!
+
+  # Unused until checkout
+  #workflow: Workflow # Needs more consideration
+
+  # Read-only: Carts that don't have an account should be considered anonymous
+  # and should be cleaned up after some time. This should not be set explicitly
+  # on the cart but can be derived and add to the Cart as a read-only
+  # attribute (read, value from a function.) The cart expiration ttl can then be
+  # modified without having to modify cart records.
+  # This value should not be stored in the database.
+  "TimeStamp at which the cart will expire."
+  expiresAt: DateTime!
+}
+
+"Wraps a list of `CartItem`s, providing pagination cursors and information."
+type CartItemConnection implements NodeConnection {
+  edges: [CartItemEdge]
+  nodes: [CartItem]
+  pageInfo: PageInfo!
+  totalCount: Int!
+}
+
+"A connection edge in which each node is a `CartItem` object"
+type CartItemEdge implements NodeEdge {
+  cursor: ConnectionCursor!
+  node: CartItem
+}
+
+"""
+A single item in a cart. The item contains information about an intended purchase.
+"""
+type CartItem implements Node {
+  "A unique identifier for the type."
+  _id: ID!
+
+  "Date first added to cart"
+  addedAt: DateTime! # Should this be "createdAt" ?
+
+  "User defined fields that can be used for any purpose."
+  metafields: [Metafield]
+
+  "Packing information such as item weight, height, length, and depth."
+  parcel: ShippingParcel # Should this be required?
```

Comment:
@aldeed I think we need parcel info. The data is required input to shipping so that it can calculate quotes. The general idea is to flatten out the cart and not include the full product record. For products, it really only needs data for display and input for the related services tax, shipping, and discounts/promotions.

Reply:
I debated this as well, let's chat about it when we sync up on this.